src/libs/ReportUtils.js

```diff
@@ -1476,26 +1476,27 @@ function buildOptimisticIOUReportAction(type, amount, currency, comment, partici
     };
 }
 
-function buildOptimisticReportPreview(reportID, iouReportID, payeeAccountID) {
+function buildOptimisticReportPreview(chatReport, iouReport) {
+    const message = getReportPreviewMessage(iouReport);
     return {
         reportActionID: NumberUtils.rand64(),
-        reportID,
-        created: DateUtils.getDBTime(),
+        reportID: chatReport.reportID,
         actionName: CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW,
         pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
-        accountID: payeeAccountID,
+        originalMessage: {
+            linkedReportID: iouReport.reportID,
+        },
         message: [
             {
-                html: '',
-                text: '',
+                html: message,
+                text: message,
                 isEdited: false,
                 type: CONST.REPORT.MESSAGE.TYPE.COMMENT,
             },
         ],
-        originalMessage: {
-            linkedReportID: iouReportID,
-        },
-        actorAccountID: currentUserAccountID,
+        created: DateUtils.getDBTime(),
+        accountID: iouReport.managerID || 0,
+        actorAccountID: iouReport.managerID || 0,
```

Comment:
For when the action is created other request or send money the actor acountID will be the person creating it so that should be correct.

As far as it goes for the default I am not sure what else we should use here

Reply:
It's expected to be 0 for the default, and it happens the case when the workspace owes money. The workspace doesn't have an `accountID` and we have logic that shows the workspace as the actor for the report preview when the `accountID` is 0.